### ISSUE TITLE
Overfitting only when using multiple processes in fit_generator

### ISSUE THREAD

**Issue by crowsonkb** [fchollet/keras#3362]
Please make sure that the boxes below are checked before you submit your issue. Thank you!

- [-] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [-] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [-] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

Script to reproduce (needs the tiny-imagenet dataset): https://gist.github.com/crowsonkb/b32e2f57bbeb7ea692fa6612fb2c02ef

This is really weird. I'm getting really bad overfitting when using ImageDataGenerator/fit_generator *only* in multiple processes mode, and *only* with more than one worker process. My accuracy during the first epoch will rise as high as 0.75 with enough worker processes, even though my validation accuracy is only around 0.01.

**Comment by crowsonkb** [comment]
I figured this one out. I was supplying a random seed for shuffling to flow_from_directory(), which caused the per-process random seed to be disregarded and the same data to be presented for training over and over. By reading the source code, I found that if you just don't specify a random seed the per-process seed will be respected. Removing the 'seed' argument fixed my problem.
